newIDE/app/src/Leaderboard/LeaderboardProvider.js

```diff
@@ -400,6 +407,40 @@ const LeaderboardProvider = ({ gameId, children }: Props) => {
     [currentLeaderboardId, displayOnlyBestEntry, fetchEntries, gameId]
   );
 
+  const previousCurrentLeaderboardId = React.useRef<?string>(null);
+  const previousCurrentLeaderboardResetLaunchedAt = React.useRef<?string>(null);
+
+  React.useEffect(
+    () => {
+      if (
+        previousCurrentLeaderboardId.current === currentLeaderboardId &&
+        !!previousCurrentLeaderboardResetLaunchedAt.current &&
+        !!currentLeaderboard &&
+        !currentLeaderboard.resetLaunchedAt
+      ) {
+        fetchEntries();
+      }
+      previousCurrentLeaderboardId.current = currentLeaderboardId;
+      previousCurrentLeaderboardResetLaunchedAt.current = currentLeaderboard
+        ? currentLeaderboard.resetLaunchedAt
+        : null;
+    },
+    [currentLeaderboard, currentLeaderboardId, fetchEntries]
+  );
+
+  useInterval(
+    () => {
+      listLeaderboards(true);
```

Comment:
last nitpick, maybe make this prop a param of an object rather than a random boolean => `listLeaderboards({shouldNotClearBeforeFetching: true})`